build-tools/packages/build-cli/src/commands/check/policy.ts

```diff
@@ -240,8 +243,7 @@ export class CheckPolicy extends BaseCommand<typeof CheckPolicy> {
 
 		handlers
 			.filter((handler) => handler.match.test(relPath))
-			// eslint-disable-next-line unicorn/no-array-for-each
-			.forEach((handler) => {
+			.map(async (handler) => {
```

Comment:
Nit: not a fan of using `map` to just iterate instead of transforming. But that segways into the more important bit: I think we need to await the list of promises returned by it or they could end up floating. A handler that yields the JS turn might not have finished executing when `map()` returns, and if the process moves on it might even terminate without waiting for the handler promise to resolve/reject.

Reply:
Map is handy here because it collects all the promises. But you're right I need to wrap it in Promise.all().